pkg/crc/preflight/preflight_checks_linux.go

```diff
@@ -0,0 +1,479 @@
+	//cmd.Env = cmdUtil.ReplaceEnv(cmd.Env, "LANG", "C")
+
+	out, err := cmd.Output()
+	if err != nil {
+		return false, nil
+	}
+
+	stdOut := string(out)
+	outputSlice := strings.Split(stdOut, "\n")
+	for _, l := range outputSlice {
+		l = strings.TrimSpace(l)
+		match, err := regexp.MatchString("^default\\s", l)
+		if err != nil {
+			return false, err
+		}
+		if match {
+			return true, nil
+		}
+	}
+	return true, nil
+}
+
+func fixDefaultPoolAvailable() (bool, error) {
+	config := poolConfig{
+		PoolName: constants.PoolName,
+		PoolDir:  constants.PoolDir,
+	}
+
+	t, err := template.New("poolxml").Parse(poolTemplate)
+	if err != nil {
+		return false, err
+	}
+	var poolXMLDef strings.Builder
+	err = t.Execute(&poolXMLDef, config)
+	if err != nil {
+		return false, err
+	}
+
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "pool-define", "/dev/stdin")
+	cmd.Stdin = strings.NewReader(poolXMLDef.String())
+	err = cmd.Run()
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func checkDefaultPoolHasSufficientSpace() (bool, error) {
+	var freeSpace poolXMLAvailableSpace
+	// Check free space in default pool
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "pool-dumpxml", "default")
+	stdOut, err := cmd.Output()
+	if err != nil {
+		return false, err
+	}
+	err = xml.Unmarshal(stdOut, &freeSpace)
+	if err != nil {
+		return false, err
+	}
+
+	size, err := units.FromHumanSize(defaultPoolSize)
+	if err != nil {
+		return false, nil
+	}
+
+	if freeSpace.Available >= float64(size) {
+		return true, nil
+	}
+	return false, fmt.Errorf("Not enough space in default pool, available free space: %s", units.HumanSize(freeSpace.Available))
+}
+
+func fixDefaultPoolHasSufficientSpace() (bool, error) {
+	return false, errors.New("Increase the size of /var, or free up space by deleting")
+}
+
+func checkLibvirtCrcNetworkAvailable() (bool, error) {
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-list")
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return false, err
+	}
+	stdOut := string(out)
+	outputSlice := strings.Split(stdOut, "\n")
+	for _, stdOut = range outputSlice {
+		stdOut = strings.TrimSpace(stdOut)
+		match, err := regexp.MatchString("^crc\\s", stdOut)
+		if err != nil {
+			return false, err
+		}
+		if match {
+			return true, nil
+		}
+	}
+	return false, errors.New("Libvirt network crc not found")
+}
+
+func fixLibvirtCrcNetworkAvailable() (bool, error) {
+	config := nodeNetworkConfig{
+		NodeMac:     constants.NodeMac,
+		NodeDomName: constants.DomName,
+		NodeIP:      constants.NodeIP,
+	}
+
+	t, err := template.New("netxml").Parse(crcNetworkTemplate)
+	if err != nil {
+		return false, err
+	}
+	var netXMLDef strings.Builder
+	err = t.Execute(&netXMLDef, config)
+	if err != nil {
+		return false, err
+	}
+	cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-define", "/dev/stdin")
+	cmd.Stdin = strings.NewReader(netXMLDef.String())
+	err = cmd.Run()
```

Comment:
Each `cmd` which is executed using the `exec` should also have `stderr`defined otherwise in case of failure it will just give the fatal but not the actual error.

Assume you have the `tt0` bridge already available and then try to create the same.

```
$ sudo ./crc setup
[sudo] password for prkumar: 
crc - Local OpenShift 4.0 clusters
INFO -- [Setting up virtualization] ...           
INFO -- [Setting up KVM] ...                      
INFO -- [Installing Libvirt] ...                  
INFO -- [Adding user to libvirt group] ...        
INFO -- [Enabling libvirt] ...                    
INFO -- [Starting Libvirt service] ...            
INFO -- [Installing docker-machine-driver-kvm] ...  
INFO -- [Creating default storage pool] ...       
INFO -- [Setting up default pool] ...             
INFO -- [Setting up Libvirt crc network] ...      
FATAL [exit status 1]  ===> This doesn't give any idea why it is failed :(
```

```golang
	cmd := exec.Command("virsh", "--connect", "qemu:///system", "net-define", "/dev/stdin")
	cmd.Stdin = strings.NewReader(netXMLDef.String())
	buf := new(bytes.Buffer)
	cmd.Stderr = buf
	err = cmd.Run()
	if err != nil {
		return false, fmt.Errorf("%v, %s", err, buf.String())
	}
```

```
$ sudo ./crc setup
[sudo] password for prkumar: 
crc - Local OpenShift 4.0 clusters
INFO -- [Setting up virtualization] ...           
INFO -- [Setting up KVM] ...                      
INFO -- [Installing Libvirt] ...                  
INFO -- [Adding user to libvirt group] ...        
INFO -- [Enabling libvirt] ...                    
INFO -- [Starting Libvirt service] ...            
INFO -- [Installing docker-machine-driver-kvm] ...  
INFO -- [Creating default storage pool] ...       
INFO -- [Setting up default pool] ...             
INFO -- [Setting up Libvirt crc network] ...      
FATAL [exit status 1, error: Failed to define network from /dev/stdin
error: internal error: bridge name 'tt0' already in use.   => this now gives why it failed.
]
```

Reply:
@praveenkumar Updated the PR, have a look